logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/MessageSendFailureHandler.kt

```diff
@@ -80,13 +83,18 @@ class MessageSendFailureHandlerImpl internal constructor(
         failure: CoreFailure,
         conversationId: ConversationId,
         messageId: String,
-        messageType: String
+        messageType: String,
+        scheduleResendIfNoNetwork: Boolean
     ) {
-        when (failure) {
-            is NetworkFailure.FederatedBackendFailure -> {
+        when {
+            failure is NetworkFailure.FederatedBackendFailure -> {
```

Comment:
Why failure was moved from when clause?

Reply:
Because the second case is not only checking the `failure` type but also `scheduleResendIfNoNetwork`:
```
failure is NetworkFailure.NoNetworkConnection && scheduleResendIfNoNetwork -> {
```